Whiskey/Functions/Resolve-WhiskeyDotNetSdkVersion.ps1

```diff
@@ -133,7 +106,277 @@ function Resolve-WhiskeyDotNetSdkVersion
+            {
+                $sortedSdkVersions =
+                    $sortedSdkVersions |
+                    Where-Object {
+                        $_.Build -eq $expectedBuild.Build
+                    }
+            }
+            $resolvedVersion =
+                $sortedSdkVersions |
+                Where-Object { $_ -in $installedVersions } |
+                Select-Object -First 1
+            if ( -not $resolvedVersion )
+            {
+                $resolvedVersion = $sortedSdkVersions | Select-Object -First 1
+            }
+        }
+        Minor
+        {
+            if ( -not $sortedSdkVersions )
+            {
+                $sortedSdkVersions = Get-AvailableSdkVersions -ReleasesIndex $releasesIndex `
+                                                              -TargetMajor $desiredVersion.Major `
+                                                              -TargetMinor ($desiredVersion.Minor + 1)
+                $expectedBuild = [Version] "0.0.1"
+            }
+            else
+            {
+                $expectedBuild = $sortedSdkVersions | Select-Object -Last 1
+            }
+
+            if ( $expectedBuild )
+            {
+                $sortedSdkVersions =
+                    $sortedSdkVersions |
+                    Where-Object { $_.Build -eq $expectedBuild.Build }
+            }
+            $resolvedVersion =
+                $sortedSdkVersions |
+                Where-Object { $_ -in $installedVersions } |
+                Select-Object -First 1
+            if ( -not $resolvedVersion )
+            {
+                $resolvedVersion = $sortedSdkVersions | Select-Object -First 1
+            }
+        }
+        Major
+        {
+            if ( -not $sortedSdkVersions )
+            {
+                $sortedSdkVersions = Get-AvailableSdkVersions -ReleasesIndex $releasesIndex `
+                                                              -TargetMajor ($desiredVersion.Major + 1)`
+                                                              -TargetMinor 0
+                $expectedBuild = [Version] "0.0.1"
+            }
+            else
+            {
+                $expectedBuild = $sortedSdkVersions | Select-Object -Last 1
+            }
+            if ( $expectedBuild )
+            {
+                $sortedSdkVersions =
+                    $sortedSdkVersions |
+                    Where-Object { $_.Build -eq $expectedBuild.Build }
+            }
+            $resolvedVersion =
+                $sortedSdkVersions |
+                Where-Object { $_ -in $installedVersions } |
+                Select-Object -First 1
+            if ( -not $resolvedVersion )
+            {
+                $resolvedVersion = $sortedSdkVersions | Select-Object -First 1
+            }
+        }
+        LatestPatch
+        {
+            $sortedSdkVersions =
+                $sortedSdkVersions |
+                Where-Object { $_.Build -eq $desiredVersion.Build }

```

Comment:
This should filter versions by the first 3 parts of the version:

```powershell
Where-Object {
  $_.Major -eq $desiredMajor -and
  $_.Minor -eq $desiredMinor -and
  $_.Build -eq $desiredBuild
}
```

Reply:
I decided to omit the first 2 parts of the version because when we grab the .NET SDK versions from Microsoft we are only able to target a specific major-minor set. These can be still be added for clarity but they are unnecessary for both the `Patch` and `Feature` branches

Reply:
Ah that makes sense now. I would still add them for clarity as it can be unclear, evidenced by comment.